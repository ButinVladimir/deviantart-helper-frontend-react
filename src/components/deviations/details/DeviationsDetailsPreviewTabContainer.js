import { connect } from 'react-redux';
import DeviationsDetailsPreviewTab from './DeviationsDetailsPreviewTab';

/**
 * @description
 * Maps Redux state to props.
 *
 * @param {State} state - The Redux state.
 * @returns {Object} Props.
 */
const mapStateToProps = state => ({
  preview: state.deviations.details.preview,
  title: state.deviations.details.title,
});

export default connect(mapStateToProps)(DeviationsDetailsPreviewTab);
