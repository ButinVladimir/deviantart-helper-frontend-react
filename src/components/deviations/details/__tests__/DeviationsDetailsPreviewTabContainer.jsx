import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import createDefaultState from '../../../../redux/states/state';
import DeviationsDetailsPreviewTabContainer from '../DeviationsDetailsPreviewTabContainer';

describe('DeviationsDetailsPreviewTabContainer', () => {
  it('can be rendered correctly', () => {
    const state = {
      ...createDefaultState(),
      deviations: {
        details: {
          preview: {
            src: '/path/to/image',
            width: 1,
            height: 2,
          },
          title: 'Title',
        },
      },
    };

    const mockStore = configureStore();
    const store = mockStore(state);
    const wrapper = mount(
      <Provider store={store}>
        <DeviationsDetailsPreviewTabContainer />
      </Provider>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
