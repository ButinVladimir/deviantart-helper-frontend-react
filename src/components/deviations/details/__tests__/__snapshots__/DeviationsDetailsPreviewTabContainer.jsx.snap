// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviationsDetailsPreviewTabContainer can be rendered correctly 1`] = `
<section
  className="section"
  style={Object {}}
>
  <div
    className="container"
    style={Object {}}
  >
    <div
      className="level"
      style={Object {}}
    >
      <div
        className="level-item"
        style={Object {}}
      >
        <img
          alt="Title"
          src="/path/to/image"
        />
      </div>
    </div>
  </div>
</section>
`;
